libqtile/state.py

```diff
@@ -45,11 +45,11 @@ def apply(self, qtile):
         Rearrange the windows in the specified Qtile object according to this
         QtileState.
         """
-        for (group, layout) in self.groups.items():
+        for (group, layout) in self.groups:
```

Comment:
I'm not sure how important it is to maintain restart-ability across versions, I'd be ok breaking that to make improvements to the way it runs.